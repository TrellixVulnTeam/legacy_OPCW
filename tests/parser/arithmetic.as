# Addition
4 + 320;
52 + 434
+ 25 + (523 + 30);

# Subtraction
43 - 43;
63 - 36 +
325;
63 - (36 + 430);

# Multiplication
43 * 43;
63 * 36 *
63 * (36 + 430);
32 + 30 * 10;

# Division
43 / 43;
63 / 36 *
63 / (36 + 430);
32 + 30 / 10;

# Modulo
43 % 43;
63 % 36 /
63 % (36 + 430);
32 + 30 % 10;
