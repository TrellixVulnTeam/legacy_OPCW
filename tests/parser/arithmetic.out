ModuleDecl <?>
  AddExpr <?>
    IntegerExpr <?> 'int' 4 (10)
    IntegerExpr <?> 'int' 320 (10)
  AddExpr <?>
    AddExpr <?>
      AddExpr <?>
        IntegerExpr <?> 'int' 52 (10)
        IntegerExpr <?> 'int' 434 (10)
      IntegerExpr <?> 'int' 25 (10)
    AddExpr <?>
      IntegerExpr <?> 'int' 523 (10)
      IntegerExpr <?> 'int' 30 (10)
  SubtractExpr <?>
    IntegerExpr <?> 'int' 43 (10)
    IntegerExpr <?> 'int' 43 (10)
  AddExpr <?>
    SubtractExpr <?>
      IntegerExpr <?> 'int' 63 (10)
      IntegerExpr <?> 'int' 36 (10)
    IntegerExpr <?> 'int' 325 (10)
  SubtractExpr <?>
    IntegerExpr <?> 'int' 63 (10)
    AddExpr <?>
      IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 430 (10)
  MultiplyExpr <?>
    IntegerExpr <?> 'int' 43 (10)
    IntegerExpr <?> 'int' 43 (10)
  MultiplyExpr <?>
    MultiplyExpr <?>
      MultiplyExpr <?>
        IntegerExpr <?> 'int' 63 (10)
        IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 63 (10)
    AddExpr <?>
      IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 430 (10)
  AddExpr <?>
    IntegerExpr <?> 'int' 32 (10)
    MultiplyExpr <?>
      IntegerExpr <?> 'int' 30 (10)
      IntegerExpr <?> 'int' 10 (10)
  DivideExpr <?>
    IntegerExpr <?> 'int' 43 (10)
    IntegerExpr <?> 'int' 43 (10)
  DivideExpr <?>
    MultiplyExpr <?>
      DivideExpr <?>
        IntegerExpr <?> 'int' 63 (10)
        IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 63 (10)
    AddExpr <?>
      IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 430 (10)
  AddExpr <?>
    IntegerExpr <?> 'int' 32 (10)
    DivideExpr <?>
      IntegerExpr <?> 'int' 30 (10)
      IntegerExpr <?> 'int' 10 (10)
  ModuloExpr <?>
    IntegerExpr <?> 'int' 43 (10)
    IntegerExpr <?> 'int' 43 (10)
  ModuloExpr <?>
    DivideExpr <?>
      ModuloExpr <?>
        IntegerExpr <?> 'int' 63 (10)
        IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 63 (10)
    AddExpr <?>
      IntegerExpr <?> 'int' 36 (10)
      IntegerExpr <?> 'int' 430 (10)
  AddExpr <?>
    IntegerExpr <?> 'int' 32 (10)
    ModuloExpr <?>
      IntegerExpr <?> 'int' 30 (10)
      IntegerExpr <?> 'int' 10 (10)
